compact_test.go

```diff
@@ -110,6 +110,105 @@ func Test_DecodeCompact_U16(t *testing.T) {
 	assert.Equal(t, expect, actual)
 }
 
+func Test_DecodeCompact_U64_Mode0(t *testing.T) {
```

Comment:
nit: seems like an ideal case for table driven tests  https://golang.cafe/blog/golang-table-test-example